src/compiler/checker.ts

```diff
@@ -14541,9 +14541,8 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
                 }
             }
 
-            const classType = declaration.kind === SyntaxKind.Constructor ?
-                getDeclaredTypeOfClassOrInterface(getMergedSymbol((declaration.parent as ClassDeclaration).symbol))
-                : undefined;
+            const constructorDeclaration = tryGetConstructorDeclaration(declaration);

```

Comment:
I'd like to inline tryGetConstructorDeclaration because
- is only two lines long and used once
- its name doesn't mention jsdoc, even though that's half of the code.

The overhead of the function declaration, especially since it needs a longer name, makes inlining it desirable.

Reply:
Removed unnecessary helper